tests/python/unittest/test_numpy_ndarray.py

```diff
@@ -999,6 +999,16 @@ def test_setitem_autograd(np_array, index):
     test_getitem_slice_bound()
 
 
+@use_np
+@pytest.mark.parametrize('load_fn', [_np.load, npx.load])
+def test_np_save_load_large_ndarrays(load_fn, tmp_path):
+    weight = mx.np.arange(32768 * 512).reshape((32768, 512))
```

Comment:
I think we can try another strange-looking shape, e.g., (32765, 511)

Reply:
The only requirement for triggering the bug is that the total size needs to be greater than the chunk size used by miniz when writing to the file

Reply:
adding a few varied sizes may help capture other bugs in the future, including those on the very small and very large ends.

Reply:
that said, I think fixing the bug is more urgent and I'm ok with merging it once CI passes

Reply:
Yes, we can add more tests later. We should always try to add more tests to these the basic functionalities.